docs/practices/appeng/adrs/0005-javascript-dependency-management.mdx

```diff
@@ -0,0 +1,273 @@
+not publish changes at all. But knowing _what_ is actually changing in the code
+when you update a package can help immensely with having the confidence that
+nothing will break, as well as knowing what part of the code might be affected
+and what to test.
+
+Another tip is to update related packages together. For example, it's often the
+case that all `@storybook` packages will need to be updated at the same time to
+maintain compatibility. `react` and `react-dom` have the same requirement. This
+is where having a reasonable understanding of _what_ the packages are and
+how/why they're used comes into play.
+
+An example update path might look like:
+
+- Update all eslint related packages/plugins
+  - Run the linter to make sure there are no new issues
+- Update all Jest/testing-library related packages
+  - Run the tests to make sure there are no new test failures
+- Update packages related to the compilation/build config (such as Babel,
+  TypeScript, etc.)
+  - Run the build to make sure it still compiles without errors
+- Update all Storybook packages
+  - Run Storybook to make sure the build still works
+- Update production dependencies grouped by feature/function/runtime
+  - i.e., `next, @opentelemetry, apollo-server`, etc.
+  - Run the application in both development & production modes, test manually
+  - Run automated tests
+
+#### What about code changes?
+
+What happens when a package update requires code changes? In theory, if
+following strict semantic versioning, patch & minor updates never should.
+However, not all packages follow semantic versioning, and also all packages are
+(probably) maintained by humans, who make mistakes. Required changes might fall
+under any of the following categories:
+
+- Small, semantic or not functionally different, documented code/API changes
+- More substantial, documented changes that might have a functional impact
+- Undocumented changes needed because of unexpected issues or edge cases
+
+When & how you make these changes can be a gut feeling based on the situation.
+If an update requires making a small change in a handful of places, that could
+be easily completed as part of the regular update cadence. If an update requires
+an easy change but one that will cause a diff in many files (such as a renamed
+import), it might make sense to isolate that on a separate branch. If an update
+requires upgrading several packages at once, and/or also making substantial code
+changes, that should probably also be done on its own.
+
+Ultimately, the goal is to minimize risk by ensuring that whatever changes you
+do make can be easily tested, and if an issue is found it's straightforward to
+either debug or roll back from. That all comes down to identifying what
+package introduced the issue, and how to resolve it. If a change involves
+updating many packages _and_ making many code changes at once, that will be
+much more difficult.
+
+### Working with files used and generated by JDMT
+
```

Comment:
This section will discuss using `.gitattributes` and `.gitignore` to prevent editing/diffing Lock files and prevent mixing those Lock files respectively. I will probably get to this at some point today or tomorrow.